module/zfs/zil.c

```diff
@@ -289,6 +292,14 @@ zil_read_log_data(zilog_t *zilog, const lr_write_t *lr, void *wbuf)
 	if (zilog->zl_header->zh_claim_txg == 0)
 		zio_flags |= ZIO_FLAG_SPECULATIVE | ZIO_FLAG_SCRUB;
 
+	/*
+	 * If we are not using the resulting data, we are just checking that
+	 * it hasn't been corrupted. We shouldn't try to decrypt it now since
+	 * the keys may not be loaded.
+	 */
+	if (wbuf == NULL && BP_IS_ENCRYPTED(bp))
+		zio_flags |= ZIO_FLAG_RAW;
```

Comment:
I don't see why not. I just didn't want to impact existing functionality. I'll run some tests to make sure that works and make the change.